sound/soc/sof/trace.c

```diff
@@ -44,8 +44,9 @@ static int trace_filter_parse_entry(struct snd_sof_dev *sdev, const char *line,
 	int ret;
 
 	/* ignore empty content */
-	if (!line[0])
-		return 0;
+	sscanf(line, " %n", &read);
+	if (read == len)
+		return len;
 
 	ret = sscanf(line, " %d %x %d %d %n", &log_level, &uuid_id, &pipe_id, &comp_id, &read);
```

Comment:
@ktrzcinx Hmm, the sscanf just to read whitespace seems a bit odd. How about if we consume the linefeed in the existing sscanf calll on L51. Would this work?

Reply:
> sscanf just to read whitespace

rather to check if input string consist only from whitespaces or is empty one, so its eg one of `"\0"`, `"\n\0"`, `" \0"`, `" \n\0"`.

> seems a bit odd

I think it shortest and reliable solution as long as reader are familiar with `scanf()` format string. I didn't use any hack here, it's just standard function usage.

> How about if we consume the linefeed in the existing sscanf calll on L51. Would this work?

There already is linefeed consumer in the existing sscanf calll on L51 and it doesn't work in all situations 😄. The case here is parsing last entry of `strsep("0 0 0 0;\n"; ";")`, it will be "\n", so scanning 4 numbers fails for it.

Reply:
@ktrzcinx Thanks. I agree this is probably the cleanest, but I'm still not entirely convinced having this sort of relaxed parsing in kernel space is the right place. I mean "0 0 0 0;\n" is an invalid input in the end. Why have extra code in kernel for something user-space can easily just get right.

Reply:
Without this change we have some inconsistence - `"0 0 0 0;\n"` returns parsing error but `"0 0 0 0\n"` passes, after change it will be consistent.

Reply:
I get the point @ktrzcinx about detecting a line with whitespaces only, but I don't get how this change helps you detect 
"0 0 0 0;\n" from "0 0 0 0\n"

Reply:
@plbossart `"0 0 0 0;\n"` will be split to `"0 0 0 0\0"` (parsed with success) and `"\n"` (rising parse error without this patch). Such and situation doesn't happen for `"0 0 0 0\n"`, because there's no spiting char, and this text will be consumed at once.